tests/testsuite.src/syn_misc.at

```diff
@@ -8385,3 +8385,105 @@ prog.cob:18: error: ANY LENGTH items may only be BY REFERENCE formal parameters
 
 AT_CLEANUP
 
+AT_SETUP([IMP INCLUDE directive])
+AT_KEYWORDS([IMP INCLUDE])
+
+AT_DATA([prog.cob], [
+       >>IMP INCLUDE "file.h"
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       PROCEDURE DIVISION.
+           GOBACK.
+])
+
+AT_CHECK([$COMPILE_ONLY -fformat=fixed prog.cob], [0], [], [])
+
+AT_DATA([prog.cob], [
+>>IMP INCLUDE "file.h"
+identification division.
+program-id. prog.
+procedure division.
+    goback.
+])
+
+AT_CHECK([$COMPILE_ONLY -fformat=free prog.cob], [0], [], [])
+
+AT_CLEANUP
+
+AT_SETUP([IMP INCLUDE directive multiple files])
+AT_KEYWORDS([IMP INCLUDE])
+
+AT_DATA([prog.cob], [
+>>IMP INCLUDE "file1.h" "file2.h" "file3.h"
+identification division.
+program-id. prog.
+procedure division.
+    goback.
+])
+
+AT_CHECK([$COMPILE_ONLY -fformat=free prog.cob], [0], [], [])
+
+AT_CLEANUP
+
+AT_SETUP([IMP INCLUDE directive file format])
+AT_KEYWORDS([IMP INCLUDE])
+
+AT_DATA([prog.cob], [
+>>IMP INCLUDE "file"
+identification division.
+program-id. prog.
+procedure division.
+    goback.
+])
+
+AT_CHECK([$COMPILE_ONLY -fformat=free prog.cob], [1], [],
+[prog.cob:2: error: invalid IMP directive option 'file'
+])
+
+AT_DATA([prog.cob], [
+>>IMP INCLUDE file.h
+identification division.
+program-id. prog.
+procedure division.
+    goback.
+])
+
+AT_CHECK([$COMPILE_ONLY -fformat=free prog.cob], [1], [],
+[prog.cob:2: error: syntax error, unexpected ., expecting Word or Literal
+prog.cob:2: error: PROGRAM-ID header missing
+prog.cob:2: error: syntax error, unexpected end of file, expecting SECTION
+])
+
+AT_DATA([prog.cob], [
+>>IMP INCLUDE "file1.h" "file2" "file3.h"
+identification division.
+program-id. prog.
+procedure division.
+    goback.
+])
+
+AT_CHECK([$COMPILE_ONLY --fformat=free prog.cob], [1], [],
+[prog.cob:2: error: invalid IMP directive option 'file2'
+])
+
+AT_DATA([prog.cob], [
+>>IMP INCLUDE "file1.h" file2 "file3.h"
+identification division.
+program-id. prog.
+procedure division.
+    goback.
+])
+
+AT_CHECK([$COMPILE_ONLY --fformat=free prog.cob], [1], [],
+[prog.cob:2: error: PROGRAM-ID header missing
+prog.cob:2: error: ENVIRONMENT DIVISION header missing
+prog.cob:2: error: CONFIGURATION SECTION header missing
+prog.cob:2: error: SPECIAL-NAMES header missing
+prog.cob:2: error: invalid system-name 'file2'
+prog.cob:2: error: invalid symbol '#' - skipping word
+prog.cob:2: error: invalid system-name 'h'
+prog.cob:4: error: syntax error, unexpected IDENTIFICATION, expecting CRT or Identifier
+prog.cob:4: error: syntax error, unexpected DIVISION
+])
```

Comment:
where's the error here?

Reply:
Since this directive is part of the syntax of programs (unlike the compiler option), there is apparently no explicit restriction. Hence, I thought of allowing only quoted file names.

Or should I be more permissive and also allow unquoted file names? Should I also assume that GnuCOBOL's philosophy is to be as permissive as possible?

Reply:
The directive should take what the command line option takes. I think that if you put garbage in there on the command line, you will get garbage during compile, so the same can apply here. In any case there should be no COBOL parser error like the one that is currently expected here